sdk/nodejs/tests/provider/experimental/analyzer.spec.ts

```diff
@@ -206,9 +209,9 @@ describe("Analyzer", function () {
                     },
                 },
                 outputs: {
-                    outArrayOfStrings: { type: "array", items: { type: "string" } },
-                    outArrayOfNumbers: { type: "array", items: { type: "number" } },
-                    outArrayOfBooleans: { type: "array", items: { type: "boolean" } },
+                    outArrayOfStrings: { type: "array", items: { type: "string", plain: true } },
+                    outArrayOfNumbers: { type: "array", items: { type: "number", plain: true } },
+                    outArrayOfBooleans: { type: "array", items: { type: "boolean", plain: true } },
```

Comment:
I think output (sub-)types can never be plain?

Reply:
I thought that's what you were suggesting in https://github.com/pulumi/pulumi/pull/18636#discussion_r1963965398?  Am I misunderstanding that? Or are inputs and outputs supposed to be different here?  

Python also seems to have this plain: https://github.com/pulumi/pulumi/blob/master/sdk/python/lib/test/provider/experimental/test_analyzer.py#L249

Reply:
My comment that you linked was about input types not outputs. I don't think that plain makes practical sense for outputs... If you look at our codegen, it generates types like `pulumi.Output<{[key: string]: string}` or `pulumi.Output<string[]>` but those aren't marked as plain in the schema. I don't know what happens if we do mark them - probably they are ignored?

Python looks wrong to me too.

Reply:
If it's just ignored, I would leave the plain here since that's much simpler implementation wise, given the difference in inputs and outputs here.  Otherwise we have to pass through whether we're dealing with an input or output everywhere which would be quite awkward.  Looking at what codegen generates for the sdk it's `pulumi.Output[Sequence[str]]` in python and `pulumi.Output<string[]>;` in typescript, so I would leave it as it is.

Reply:
>  we have to pass through whether we're dealing with an input or output everywhere 

I think this is exactly what I do in Java/.NET…

 In my mind, our schemas have to be precise. I imagine us writing schemas for conformance tests, generating classes from them, then inferring schemas to validate that the round trip is lossless. Any divergence is problematic then.

Also, it’s hard to prove that no downstream schema client would not be affected by this divergence… Codegen? Programgen? Docs? Schema diff? Cosmic’s tools?

Reply:
Ok, I've changed this in the latest commit.

Reply:
Opened an issue for python as well, and added it to the epic: https://github.com/pulumi/pulumi/issues/18694.  Not sure it needs to be in this iteration, but we can discuss later.